bundles/io/org.eclipse.smarthome.io.security/src/main/java/org/eclipse/smarthome/io/security/api/Cryptographic.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.security.api;
+
+import java.security.GeneralSecurityException;
+
+/**
+ * Implement this in {@link TlsNetworkServer} to get notifications of changed
+ * certificate contexts.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public interface Cryptographic {
+    /**
+     * Encrypts the given text by the given password.
+     *
+     * @param password The password as byte array.
+     * @param plain The plain text to encrypt
+     * @return Returns the ciphered text.
+     * @throws GeneralSecurityException
+     */
+    char[] encrypt(byte[] password, char[] plain) throws GeneralSecurityException;
+
+    /**
+     * Decrypts the given text with the given password.
+     *
+     * @param password The password as byte array.
+     * @param cipher The cipher to decrypt
+     * @return Returns the plain text
+     * @throws GeneralSecurityException
+     */
+    char[] decrypt(byte[] password, char[] cipher) throws GeneralSecurityException;
```

Comment:
Cipher? Does cipher means here encrypted password? Cipher term is normally used as a term for used algorithm.

Reply:
He probably means `cipherText`. But I agree, this name choice might be confusing.

Reply:
encryptedDate would be an option

Reply:
encryptedDat**e** implies that it is a date from a calendar, so I would rather use encrytedDat**a**